### DISCUSSION TITLE
Why the separate swap space creation

### DISCUSSION THREAD

**Discussion by martdj** [HCL-TECH-SOFTWARE/connections-automation#111] [Q&A]
When running the playbook to install DB2 the swap space partition that I had created on my OS is disabled and a separate swapfile is created (which is slightly larger than the swap space partition btw). Why does the script create a separate swapfile if a swap partition is already available? Is this because of the size or does it always do this?

**Comment by dejo1307** [comment]
Hey, sorry, I'm on vacation and just got the chance to read this stuff.

It was added to specifically control amount of swap space and not rely on if swap is already on (or not) and whatever size it is (added specifically when Oracle support was added because Oracle is quite picky about swap when running prereq checks).

**Reply by martdj** [reply]
Currently the swapoff is not permanent (the swap file is not commented out in the etc/fstab file), so it leads to 2 swap spaces if a swap space was already available. It would be neat if the script could check if a swap space of big enough size is already available.
I did see that if I add a file called swapfile.ansible.success to the root of the destination server, the script will refrain from messing with the swap file, so that's a good workaround.

**Reply by dejo1307** [reply]
Oh, OK, that's a good catch - and I would say a bug on our side. I see what you are saying, will fix it once I am back (basically - if swap is already there, don't do anything and don't break anything; if there is no swap and it should be there, add it if it's not there). Thanks for pointing it out.

**Reply by dejo1307** [reply]
Pushed the fix, and it is part of the latest release (0.4.1). Should be more intelligent now, but will still take some more look to try to figure out if there is something more we can do about this.
